script.skin.helper.colorpicker/resources/lib/ColorPicker.py

```diff
@@ -0,0 +1,363 @@
+        self.action_exitkeys_id = [10, 13]
+        self.win = xbmcgui.Window(10000)
+        self.build_colors_list()
+        self.result = -1
+
+        # check paths
+        if xbmcvfs.exists(SKINCOLORFILE) and not xbmcvfs.exists(SKINCOLORFILES_PATH):
+            xbmcvfs.mkdirs(SKINCOLORFILES_PATH)
+        if not xbmcvfs.exists(COLORFILES_PATH):
+            xbmcvfs.mkdirs(COLORFILES_PATH)
+
+    def add_color_to_list(self, colorname, colorstring):
+        '''adds the coloroption as listitem to the list'''
+        if not colorname:
+            colorname = colorstring
+        color_image_file = self.create_color_swatch_image(colorstring)
+        listitem = xbmcgui.ListItem(label=colorname)
+        listitem.setArt({'icon': color_image_file})
+        listitem.setProperty("colorstring", colorstring)
+        self.colors_list.addItem(listitem)
+
+    def build_colors_list(self):
+        '''
+            build the list of colorswatches we want to display, check if skinner
+            overrides the default provided colorsfile and pick the right colorpalette
+        '''
+
+        # prefer skin colors file
+        if xbmcvfs.exists(SKINCOLORFILE):
+            colors_file = SKINCOLORFILE
+            self.colors_path = SKINCOLORFILES_PATH
+        else:
+            colors_file = os.path.join(ADDON_PATH, 'resources', 'colors', 'colors.xml')
+            self.colors_path = COLORFILES_PATH
+
+        doc = parse(colors_file)
+        palette_listing = doc.documentElement.getElementsByTagName('palette')
+        if palette_listing:
+            # we have multiple palettes specified
+            for item in palette_listing:
+                palette_name = item.attributes['name'].nodeValue
+                self.all_colors[palette_name] = self.get_colors_from_xml(item)
+                self.all_palettes.append(palette_name)
+        else:
+            # we do not have multiple palettes
+            self.all_colors["all"] = self.get_colors_from_xml(doc.documentElement)
+            self.all_palettes.append("all")
+
+    def load_colors_palette(self, palette_name=""):
+        '''load preferred color palette'''
+        self.colors_list.reset()
+        if not palette_name:
+            # just grab the first palette if none specified
+            palette_name = self.all_palettes[0]
+        # set window prop with active palette
+        if palette_name != "all":
+            self.current_window.setProperty("palettename", palette_name)
+        if not self.all_colors.get(palette_name):
+            log_msg("No palette exists with name %s" % palette_name, xbmc.LOGERROR)
+            return
+        for item in self.all_colors[palette_name]:
+            self.add_color_to_list(item[0], item[1])
+
+    def onInit(self):
+        '''Called after initialization, get all colors and build the listing'''
+        with busy_dialog():
+            xbmc.executebuiltin("ActivateWindow(busydialognocancel)")
```

Comment:
Not needed, you're activating the window twice

Reply:
thx, missed that.

Reply:
Only this one missing for the add-on to go in. Please hit "resolve conversation" as you go along the changes, if not asking too much ;)

Reply:
Aye, wasn't sure if this should be done by you reviewers or me :)